sdlib/d/gc/tcache.d

```diff
@@ -274,6 +281,67 @@ public:
 	}
 
 private:
+
+	CapacityInfo getAllocInfo(PageDescriptor pd, void* ptr) {
+		auto e = pd.extent;
+
+		if (e.isLarge()) {
+			return CapacityInfo(e.address, e.size, e.usedCapacity);
+		}
+
+		// Slab alloc:
+		import d.gc.slab;
+		auto sg = SlabAllocGeometry(ptr, pd);
+
+		// If freespace flag is 0, or this size class does not support meta,
+		// then the alloc is reported to be fully used:
+		if (!e.hasFreeSpace(sg.index)) {
+			return CapacityInfo(sg.address, sg.size, sg.size);
+		}
+
+		// Decode freesize, found in the final byte (or two bytes) of the alloc:
+		auto freeSize =
+			readPackedFreeSpace(cast(ushort*) sg.address + sg.size - 2);
+
+		return CapacityInfo(sg.address, sg.size, sg.size - freeSize);
+	}
```

Comment:
See, this is not using anything from the thread cache, so https://github.com/snazzy-d/sdc/pull/297#discussion_r1323224258 was indeed the right question to ask.